pennylane/ops/qubit.py

```diff
@@ -1977,6 +1977,228 @@ def adjoint(self, do_queue=False):
+                0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\
+                0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\
+                0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 0 & 0 & 0 & 0 & 0 & 0\\
+                0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 0 & 0 & 0 & 0 & 0\\
+                0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 0 & 0 & 0 & 0\\
+                0 & 0 & 0 & \sin(\phi/2) & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & \cos(\phi/2) & 0 & 0 & 0\\
+                0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 0 & 0\\
+                0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 0\\
+                0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1
+            \end{bmatrix}.
+
+
+    This operation performs a rotation in the two-dimensional subspace :math:`\{|1100\rangle,
+    |0011\rangle\}`. The name originates from the occupation-number representation, where
+    the transformation from :math:`|1100\rangle` to :math:`|0011\rangle` is interpreted as
+    "exciting" two particles from the first pair of qubits to the second pair of qubits.
+
+    **Details:**
+
+    * Number of wires: 4
+    * Number of parameters: 1
+    * Gradient recipe: Obtained from its decomposition in terms of the
+      :class:`~.DoubleExcitationPlus` and :class:`~.DoubleExcitationMinus` operations
+
+    Args:
+        phi (float): rotation angle :math:`\phi`
+        wires (Sequence[int]): the wires the operation acts on
+
+
+    **Example**
+
+    The following circuit performs the transformation :math:`|1100\rangle\rightarrow \cos(
+    \phi/2)|1100\rangle -\sin(\phi/2)|0011\rangle)`:
+
+    .. code-block::
+
+        @qml.qnode(dev)
+        def circuit(phi):
+            qml.PauliX(wires=0)
+            qml.PauliX(wires=1)
+            qml.DoubleExcitation(phi, wires=[0, 1, 2, 3])
+    """
+
+    num_params = 1
+    num_wires = 4
+    par_domain = "R"
+    grad_method = "A"
+
+    G = np.zeros((16, 16), dtype=np.complex64)
+    G[3, 12] = -1j  # 3 (dec) = 0011 (bin)
+    G[12, 3] = 1j  # 12 (dec) = 1100 (bin)
+    generator = [G, -1 / 2]
+
+    @classmethod
+    def _matrix(cls, *params):
+        theta = params[0]
+        c = math.cos(theta / 2)
+        s = math.sin(theta / 2)
+
+        U = np.eye(16)
+        U[3, 3] = c  # 3 (dec) = 0011 (bin)
+        U[3, 12] = -s  # 12 (dec) = 1100 (bin)
+        U[12, 3] = s
+        U[12, 12] = c
+
+        return U
+
+    @staticmethod
+    def decomposition(theta, wires):
+        decomp_ops = [
+            DoubleExcitationPlus(theta / 2, wires=wires),
+            DoubleExcitationMinus(theta / 2, wires=wires),
+        ]
+        return decomp_ops
+
+
+class DoubleExcitationPlus(Operation):
+    r"""DoubleExcitationPlus(phi, wires)
+    Double excitation rotation with positive phase-shift outside the rotation subspace.
+
+    .. math:: U_+(\phi) = \begin{bmatrix}
```

Comment:
From my browser, the matrix seems to be too big when rendered in the docs:
https://pennylane--1123.org.readthedocs.build/en/1123/code/api/pennylane.DoubleExcitationPlus.html?highlight=doubleexcitationplus#pennylane.DoubleExcitationPlus